### PR TITLE
fix(opensuse): fixes config files generation in opensuse

### PR DESCRIPTION
**Pull request by fzipi**
Signed-off-by: Felipe Zipitria <fzipitria@perceptyx.com>

### PR progress checklist (to be filled in by reviewers)

- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [x] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Related issues and/or pull requests

Issue #111 .

### Describe the changes you're proposing

Creates file needed by opensuse to prevent system changing config files after salt has created them.

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

Tests will be merged in #108 after merging this one.

### CODE REVIEW

File: postfix/init.sls [hunk 1]
```diff
@@ -1,5 +1,16 @@
 {% from "postfix/map.jinja" import postfix with context %}
 
+{%- if grains['os_family']=="Suse" %}
```
Comment (myii) [inline postfix/init.sls:3]: I see this is based on a line further below but I would suggest using the usual spacing (for that line as well):

```suggestion
{%- if grains['os_family'] == "Suse" %}
```

We generally use `grains.os_family` but keeping it as `grains['os_family']` is fine as well.

File: postfix/init.sls [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+# The existence of this file prevents the system to
+# overwrite files from salt when installing.
+/var/adm/postfix.configured:
+  file.managed:
+    - contents: ''
+    - mode: '0644'
+    - user: 'root'
+    - group: 'root'
```
Comment (myii) [inline postfix/init.sls:11]: This state needs a `require_in` (for `pkg: postfix`, right?)

File: postfix/init.sls (hunk 1 above)
Comment (fzipi) [reply postfix/init.sls]: I used the same syntax that the file had below. But it makes sense to change both then.

File: postfix/init.sls (hunk 2 above)
Comment (fzipi) [reply postfix/init.sls:11]: Exactly, good catch.

**Comment by myii** [comment]
Merged, thanks @fzipi.

**Pull request merged**